### ISSUE TITLE
usize ints and ARM Compilation break the library

### ISSUE THREAD

**Issue by ControlCplusControlV** [matter-labs/zksync#304]
To get a better understanding of this - I was compiling golem which uses this library for ARM, in hopes of running it on a Raspberry Pi. I was using - target - target.arm-unknown-linux-musleabi linker - arm-linux-gnueabihf-ld for my target and linker. 

When doing this I'd run into various issues in this library caused with integers being different lengths, which appeared to be from usize ints basing int size off of architecture, thus not working well with ARM.

I raised this issue on stackoverflow where I go a little more in depth on the issue - https://stackoverflow.com/questions/67013986/usize-breaking-zksync-when-compiling-for-arm

This helpful answer suggested I put this here - 

stat is a Statvfs structure, the return type of Statvfs::blocks_available() is fsblkcnt_t, and the return type of Statvfs::fragment_size() is c_ulong. These two types are defined in the libc package, which is a paper-thin wrapper around the low-level C OS-calls. The types are equivalent to the C types in the OS-specific *.h files. The sizes of the types vary from platform to platform.

The library you're compiling appears to have some assumptions baked in about these sizes and their arithmetic compatibility.

So in short, fix ARM compilation for this library by reworking how int lengths are handled, if this can't be done in one body of code to support all platforms, forking out may be necessary to work with ARM.

**Comment by popzxc** [comment]
Thanks for the report! Unfortunately, we currently consider 64-bit linux as the main target and have no plans in making the code compatible with Raspberry Pi.

However, if you are willing to help, you can file a PR with a fix, and we'll be happy to merge it.

**Comment by shamatar** [comment]
@popzxc Even though we do not aim to support anything other than x86-64 at the moment I think we can change a signature of the function and casts in this expression to usize as it's logical not to have more countable elements then assessable memory can handle

**Comment by Deniallugo** [comment]
Thanks for the report! The task was created in the internal tracker.
